master/internal/workspace/postgres_workspace.go

```diff
@@ -132,3 +133,41 @@ func AllWorkspaces(ctx context.Context) ([]*model.Workspace, error) {
 	}
 	return w, nil
 }
+
+// GetNamespaceFromWorkspace returns the namespace for the given workspace and cluster.
+func GetNamespaceFromWorkspace(ctx context.Context, workspaceName string, clusterName string) (string, error) {
```

Comment:
these functions need tests -- i know the eng teams have a greater reliability effort to decrease SQL errors and part of that is providing better test coverage on bun functions

Reply:
done